internal/dag/gatewayapi_processor.go

```diff
@@ -315,11 +354,106 @@ func selectorMatches(selector *metav1.LabelSelector, objLabels map[string]string
 	return true, nil
 }
 
+func (p *GatewayAPIProcessor) computeTLSRoute(route *gatewayapi_v1alpha1.TLSRoute) {
+
+	routeAccessor, commit := p.dag.StatusCache.ConditionsAccessor(k8s.NamespacedNameOf(route), route.Generation, status.ResourceTLSRoute, route.Status.Gateways)
+	defer commit()
+
+	for _, rule := range route.Spec.Rules {
+		var hosts []string
+		var matchErrors []error
+		totalSnis := 0
+
+		// Build the set of SNIs that are applied to this TLSRoute.
+		for _, match := range rule.Matches {
+			for _, snis := range match.SNIs {
+				totalSnis++
+				if err := validHostName(string(snis)); err != nil {
+					matchErrors = append(matchErrors, err)
+					continue
+				}
+				hosts = append(hosts, string(snis))
+			}
+		}
+
+		// If there are any errors with the supplied hostnames, then
+		// add a condition to the route.
+		for _, err := range matchErrors {
+			routeAccessor.AddCondition(status.ConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, err.Error())
+		}
+
+		// If all the supplied SNIs are invalid, then this route is invalid
+		// and should be dropped.
+		if len(matchErrors) != 0 && len(matchErrors) == totalSnis {
+			continue
+		}
+
+		// If SNIs is unspecified, then all
+		// requests associated with the gateway TLS listener will match.
+		// This can be used to define a default backend for a TLS listener.
+		if len(hosts) == 0 {
+			hosts = []string{"*"}
+		}
+
+		if len(rule.ForwardTo) == 0 {
+			routeAccessor.AddCondition(status.ConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, "At least one Spec.Rules.ForwardTo must be specified.")
+			continue
+		}
+
+		var proxy TCPProxy
+		for _, forward := range rule.ForwardTo {
```

Comment:
maybe `ForwardTo` validation could be abstracted to a common helper?

Reply:
👍